src/imap/mod.rs

```diff
@@ -657,17 +658,20 @@ impl Imap {
         context.sql.set_raw_config(context, &key, Some(&val)).ok();
     }
 
+    /// Fetches a single message by server UID.
+    ///
+    /// The function returns an error if the caller should try over
+    /// again later. If it succeeds the message should be treated as
+    /// received and the caller should not try to read the message
+    /// again even if no database entries are returned.
```

Comment:
Cursory reading this it reads a bit odd.  nil is returned but this talks about database entries being returned.  Could this be described differently?

Reply:
maybe first describe the Ok-case, then the error one?

Reply:
Updated the text.